src/disableActEnvironment.ts

```diff
@@ -0,0 +1,171 @@
+import {getConfig} from '@testing-library/dom'
+
+const dispose: typeof Symbol.dispose =
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  Symbol.dispose ?? Symbol.for('nodejs.dispose')
+
+export interface DisableActEnvironmentOptions {
+  /**
+   * If `true`, all modifications of values set by `disableActEnvironment`
+   * will be prevented until `cleanup` is called.
+   *
+   * @default true
+   */
+  preventModification?: boolean
+
+  /**
+   * If `true`, will change the configuration of the testing library to
+   * prevent auto-wrapping e.g. `userEvent` calls in `act`.
+   *
+   * @default true
+   */
+  adjustTestingLibConfig?: boolean
```

Comment:
```suggestion
  patchTestingLibConfig?: boolean
```

Perhaps just me, but I prefer "patch" instead of "adjust" since its a bit more technical term.

Reply:
`patch` would imply to me "we do something creepy with internals that was not intended", while this really just sets two config options. 
I want to have this sound a little more harmless :)